src/Plugin/AddHostPlugin.php

```diff
@@ -52,8 +52,11 @@ public function __construct(UriInterface $host, array $config = [])
     public function handleRequest(RequestInterface $request, callable $next, callable $first)
     {
         if ($this->replace || $request->getUri()->getHost() === '') {
-            $uri = $request->getUri()->withHost($this->host->getHost());
-            $uri = $uri->withScheme($this->host->getScheme());
+            $uri = $request->getUri()
+                ->withHost($this->host->getHost())
+                ->withScheme($this->host->getScheme())
+                ->withPort($this->host->getPort())
```

Comment:
Agreed, we should assume that the PSR-7 implementation follows PSR7, even though it is weird at some places.